File: src/pulseCounter.ts
export type PulseState = Readonly<{
  count: number;
}>;

export function initialPulseState(): PulseState {
  return { count: 0 };
}

export function recordPulse(state: PulseState): PulseState {
  return { count: state.count + 1 };
}

export function resetPulse(): PulseState {
  return initialPulseState();
}

export function getPulseMessage(count: number): string {
  if (count === 0) {
    return "Ready when you are";
  }
  if (count < 5) {
    return "Nice start";
  }
  if (count < 10) {
    return "Keep the rhythm going";
  }
  return "PocketPulse is working";
}

File: PocketPulse/PulseCounter.swift
import Foundation

struct PulseCounter: Equatable {
    private(set) var count: Int

    init(count: Int = 0) {
        self.count = max(0, count)
    }

    mutating func recordPulse() {
        count += 1
    }

    mutating func reset() {
        count = 0
    }

    var message: String {
        switch count {
        case 0:
            return "Ready when you are"
        case 1..<5:
            return "Nice start"
        case 5..<10:
            return "Keep the rhythm going"
        default:
            return "PocketPulse is working"
        }
    }
}
